File: Codility/FrogRiverOne.py
def solution(X, A):
    sol = [i for i in range(1,X+1)]
    now = []
    for i in range(len(A)):
        if A[i] not in now:
            now.append(A[i])
        if sorted(now) == sol:
            return i
    return -1


print(solution(5, [1, 3, 1, 4, 2, 3, 5, 4]))

File: Codility/FrogRiverOne.go
package main

import (
	"fmt"
)

func main() {
	arr := []int{1,3,1,4,2,3,5,4}
	fmt.Println(Solution(5, arr))
}

func Solution(X int, A []int) int {
	for idx, value := range A {
		if value >= X {
			return idx
		}
	}
	return -1
}
